packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-target-group.ts

```diff
@@ -147,7 +147,7 @@ export class ApplicationTargetGroup extends TargetGroupBase implements IApplicat
 
       if (props.stickinessCookieDuration) {
         this.enableCookieStickiness(props.stickinessCookieDuration, props.stickinessCookieName);
-      } else {
+      } else if (this.targetType !== TargetType.LAMBDA) {
```

Comment:
I don't think this would be effective with your use case. Line 157 would call the function that sets the target type based on the target

Reply:
Just to clarify what @peterwoodworth  is saying here -- this fixes the functionality when the `targetType` is explicitly defined as an input prop, but not when the type is not defined at creation, and is instead set by the target itself. Here's an example test that breaks with this implementation, but should work:

```ts
  test('Lambda target should not have stickiness.enabled set', () => {
    const app = new cdk.App();
    const stack = new cdk.Stack(app, 'Stack');
    const tg = new elbv2.ApplicationTargetGroup(stack, 'TG');
    tg.addTarget({
      attachToApplicationTargetGroup(_targetGroup: elbv2.IApplicationTargetGroup): elbv2.LoadBalancerTargetProps {
        return {
          targetType: elbv2.TargetType.LAMBDA,
          targetJson: { id: 'arn:aws:lambda:eu-west-1:123456789012:function:myFn' },
        };
      },
    });

    expect(stack).not.toHaveResourceLike('AWS::ElasticLoadBalancingV2::TargetGroup', {
      TargetGroupAttributes: [
        {
          Key: 'stickiness.enabled',
          Value: 'false',
        },
      ],
    });
  });
```

I think unfortunately we're going to have to do something kind of ugly, like detect when a Lambda targetType is added, and then unset the attribute if so (in addition to what you've done here).

Reply:
@peterwoodworth @njlynch Sorry for the late reply, and thank you for your reviews! Will take a look.